src/ark/utils/masking_utils.py

```diff
@@ -0,0 +1,123 @@
+        max_hole_size (int): maximum size of holes to leave in masked objects
+    """
+    # check correct image directory path
+    io_utils.validate_paths([img_dir])
+    fovs = io_utils.list_folders(img_dir)
+
+    # check valid channel name
+    channel_list = io_utils.remove_file_extensions(
+        io_utils.list_files(os.path.join(img_dir, fovs[0])))
+    misc_utils.verify_in_list(input_channels=channels, all_channels=channel_list)
+
+    # create composite image (or read in single image)
+    composite_imgs = composite_builder(
+        img_dir, img_sub_folder='', fov_list=fovs, images_to_add=channels, images_to_subtract=[],
+        image_type='total', composite_method='total')
+
+    for fov in fovs:
+        # create mask
+        img = composite_imgs[fov]
+        img_size = img.shape[0] * img.shape[1]
+        mask = _create_object_mask(img, 'blob', sigma, intensity_thresh_perc, max_hole_size,
+                                   fov_dim=400, min_object_area=min_mask_size,
+                                   max_object_area=img_size)
+
+        # save mask
+        save_dir = os.path.join(mask_dir, fov)
+        if not os.path.exists(save_dir):
+            os.makedirs(save_dir)
+
+        data_utils.save_fov_mask(mask_name, save_dir, mask)
+
+
+def create_cell_mask(seg_mask, cell_table, fov_name, cell_types, cluster_col, sigma,
+                     min_mask_size=0, max_hole_size=1000):
+    """Generates a binary from the cells listed in `cell_types`
+
+    Args:
+        seg_mask (numpy.ndarray): segmentation mask
+        cell_table (pandas.DataFrame): cell table containing segmentation IDs and cell types
+        fov_name (str): name of the fov to process
+        cell_types (list): list of cell types to include in the mask
+        cluster_col (str): column in cell table containing cell cluster
+        sigma (float): sigma for gaussian smoothing
+        min_mask_size (int): minimum size of a mask to include, default 0
+        max_hole_size (int): maximum size of a hole to leave without filling, default 0
+
+    Returns:
+        numpy.ndarray: binary mask
+    """
+    # get cell labels for fov and cell type
+    cell_subset = cell_table[cell_table['fov'] == fov_name]
+    cell_subset = cell_subset[cell_subset[cluster_col].isin(cell_types)]
+    cell_labels = cell_subset['label'].values
+
+    # create mask for cell type
+    cell_mask = np.isin(seg_mask, cell_labels).astype(np.int32)
+    img_size = cell_mask.shape[0] * cell_mask.shape[1]
+
+    # binarize and blur mask, no minimum size requirement or hole removal for cell masks
+    cell_mask = _create_object_mask(cell_mask, 'blob', sigma, None, max_hole_size,
+                                    fov_dim=0, min_object_area=min_mask_size,
+                                    max_object_area=img_size)
+    cell_mask[cell_mask > 0] = 1
+
+    return cell_mask
+
+
+def generate_cell_masks(seg_dir, mask_dir, cell_table, cell_types, cluster_col, mask_name,
```

Comment:
I would add a few defaults here:

- `cell_types`: default to None, in which case include all cell types in `create_cell_masks`
- `cluster_col`: should be ark.settings.cell_type. In that way you may not have to explicitly specify it in the notebook.